### DISCUSSION TITLE
Vec<u8> and JsValue

### DISCUSSION THREAD

**Discussion by jcdc404** [wasm-bindgen/wasm-bindgen#3396] [Q&A]
I am trying to console log a Vec<u8>, but I can't figure out how to convert to a JsValue so it can be logged. I see in the issues that this is resolved several years ago, but I don't see specific information on exactly how this should/could be done.  JsValue::from is not implemented for Vec<u8> or &[u8].

**Comment by JSmithOner** [comment]
have you tried:

`extern "C" {
    #[wasm_bindgen(js_namespace = console)]
    fn log(s: &str);
    #[wasm_bindgen(js_namespace = console)]
    fn warn(s: &str);
}`

`macro_rules! console_log {
  ($($t:tt)*) => (log(&format_args!($($t)*).to_string()))
}`

`console_warn!("my js value {:?}", myJsValue );`
